rust/theoros/src/rpc/pragma_feeds_registry.rs

```diff
@@ -0,0 +1,31 @@
+use pragma_utils::conversions::starknet::felt_vec_to_vec_string;
+use starknet::{
+    core::types::{BlockId, BlockTag, Felt, FunctionCall},
+    macros::selector,
+    providers::Provider,
+};
+
+use super::StarknetRpc;
+
+const PENDING_BLOCK: BlockId = BlockId::Tag(BlockTag::Pending);
+
+#[async_trait::async_trait]
+pub trait PragmaFeedsRegistryCalls {
+    /// Retrieves all the available feed ids from the Pragma Feeds Registry.
+    async fn get_feed_ids(&self, pragma_feeds_registry_address: &Felt) -> anyhow::Result<Vec<String>>;
+}
+
+#[async_trait::async_trait]
+impl PragmaFeedsRegistryCalls for StarknetRpc {
+    async fn get_feed_ids(&self, pragma_feeds_registry_address: &Felt) -> anyhow::Result<Vec<String>> {
+        let call = FunctionCall {
+            contract_address: *pragma_feeds_registry_address,
+            entry_point_selector: selector!("get_all_feeds"),
+            calldata: vec![],
+        };
+
+        let mut response = self.0.call(call, PENDING_BLOCK).await?;
+        response.insert(0, Felt::from(1)); // to fit the format given in the function
```

Comment:
no idea, was already there 😅

Reply:
@JordyRo1 any idea?

Reply:
`felt_vec_to_vec_string` requires the count of the sub array as first element. Since it's not provided in the original response, we needed to add it. Since the response result is supposed to be an Array of FeedId, the size of the subresult will always be 1.